src/main/java/core/basesyntax/figures/Circle.java

```diff
@@ -0,0 +1,39 @@
+package core.basesyntax.figures;
+
+import java.util.Locale;
+
+public class Circle extends Figure {
+    private double radius;
+    private double area;
+
+    public Circle(String color, double radius) {
+        super(color);
+        this.radius = radius;
+        this.area = calculateArea();
+    }
+
+    public double getRadius() {
+        return radius;
+    }
+
+    public void setRadius(double radius) {
+        this.radius = radius;
+        this.area = calculateArea();
+    }
+
+    public double getArea() {
+        return area;
+    }
+
+    @Override
+    protected double calculateArea() {
+        return Math.PI * radius * radius;
```

Comment:
All magic numbers in your code should be constants. The value of PI is a constant and should be defined as such, although 'Math.PI' is already a constant in Java, so you can use it directly without redefining it.